### DISCUSSION TITLE
Welcome to XrayPhysics Discussions!

### DISCUSSION THREAD

**Discussion by kylechampley** [kylechampley/XrayPhysics#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by 123xingchenn** [comment]
![屏幕截图 2024-10-15 151440](https://github.com/user-attachments/assets/de9bd9a1-24f4-4e8a-98b0-0594e1210012)

I would like to ask, is this module no longer available?

**Comment by kylechampley** [comment]
Sorry, I don't understand your question.  What module are you referring to?

**Reply by 123xingchenn** [reply]
![屏幕截图 2024-10-16 092433](https://github.com/user-attachments/assets/275f8ce2-28f6-44e5-850c-20d956752cb4)
Thanks for the reply. Specifically, in the beam hardening document, xrayphysics.xrayPhysics.setTwoMaterialBHClookupTable, this module is not available in my python.

**Reply by 123xingchenn** [reply]
My usage is still wrong, please help me.

**Comment by kylechampley** [comment]
Could you send me your Python script?

**Comment by 123xingchenn** [comment]
![Uploading image.png…]()

**Reply by 123xingchenn** [reply]
![屏幕截图 2024-10-16 134201](https://github.com/user-attachments/assets/870b3388-018a-4349-bcf5-5b78c5932212)That's probably what it looks like

**Comment by kylechampley** [comment]
Ya, the way you are importing and setting up this library is incorrect.  Did you see the examples [here](https://github.com/kylechampley/XrayPhysics/blob/main/demo/xrayphysics_example.py).

There are also examples of how to use it in LEAP.  See [d17_beam_hardening.py](https://github.com/LLNL/LEAP/blob/main/demo_leapctype/d17_beam_hardening.py) and [d18_multi-materialBHC.py](https://github.com/LLNL/LEAP/blob/main/demo_leapctype/d18_multi-materialBHC.py)

In addition, the arguments to setBHlookupTable are incorrect.  Please refer to the examples provided and the documentation.  And finally, your spectrum has negative values which doesn't make any sense.

**Reply by 123xingchenn** [reply]
Thank you, boss, I know where the problem occurs.

**Comment by Charlesgu2016** [comment]
Hi, Kyle.

Assume that kV_L and kV_H are the same, for example, 140kv, and the structure of the sandwich detector (dual-layer). Only s_L and s_H respond differently due to different filters and detector thicknesses. For example: s_L passes through 0.2 cm of Al, while s_H passes through 0.2 cm thick Al, 0.01 cm thick Gos (the thickness of the low-energy detector crystal material), and 0.05 cm thick Cu. The time to generate the DED LUT will be longer, and the results appear to be incorrect. What could be the reason for this?
![question](https://github.com/user-attachments/assets/4bf1d9e9-6d54-4a74-8118-7046690207fb)
![results](https://github.com/user-attachments/assets/143fdce3-17d8-4ec6-8f1a-34ec0879376e)
![log](https://github.com/user-attachments/assets/2b6e65ce-a47c-439d-8122-df0991e48bc1)
![speak](https://github.com/user-attachments/assets/a432c4e1-bef2-4ab4-9309-eca9b0ddac33)

**Comment by kylechampley** [comment]
This is very strange.  I was able to reproduce the issue and yes, this is a bug.

If you change the low energy kV to 139, it works fine.  I'll look into this issue.

BTW, I think you have a typo in the definition of filtResp_H.  The thickness should be 0.04 because this is the thickness of your low energy detector.

**Reply by Charlesgu2016** [reply]
yes.  you are right. Gos for low energy is 0.04cm.

**Comment by kylechampley** [comment]
I believe I fixed the bug.  Pull the latest code from the main branch to get the fix.
